### DISCUSSION TITLE
How to change perf event in perf record with Gramine

### DISCUSSION THREAD

**Discussion by RoKu-1999** [gramineproject/gramine#1442] [General]
Hello,

i have a question regarding perf record. Perf record works correctly with a gramine debug build and in debug enclave mode. 
Also, in my manifest template I have set sgx.profile.enable = "main" which writes my perf data to sgx-perf.data file.

Attached my header of the report output. My question is, usually with the -e flag of perf record you can change the event type to have a more detailed insight into cache-misses, instructions, cpu-cycles instead of the default cpu-clock event. Is there a possibility to set this flag for gramine execution aswell, as the perf data is generated by the manifest template flag and not by running perf record by urself?

This question refers to line 'Samples: 72K of event cpu-clock:HGu" in my attached out
<img width="1198" alt="Bildschirmfoto 2023-07-05 um 16 38 06" src="https://github.com/gramineproject/gramine/assets/44587538/bbcc7b23-70e8-441a-a5ae-46f4a262715e">

Thanks!

**Comment by dimakuv** [comment]
Simple answer: no, this is **not supported**.
